new c.pest file

williamaguessy opened the pull request:
I tried to ad the possibility of having "multiligne string" on c.pest. It works but after the first string detection in the pest editor, the parser is not able to recognise the comment anymore.

bjones1 commented:
- [ ] Remove commits of `Cargo.lock` or `package.lock`.
- [ ] Send this PR to the `pest` branch, instead of the `main` branch.

williamaguessy commented:
I put my change into the python pest file, for now it can well see inline comment in multiligne string as code block in this case ''' # this in not a comment ''', but not is this case : print(''' # this is not a comment '''). I aslo tried to revert my previous wrong commits.

bjones1 commented on server/src/lexer/pest_parser.rs:
```diff
@@ -231,7 +231,7 @@ mod test {
     use crate::lexer::{CodeDocBlock, DocBlock};
```
Revert these changes.

bjones1 commented on server/src/lexer/pest/python.pest:
```diff
@@ -54,34 +54,37 @@ inline_comment_contents = { logical_line ~ newline_eoi }
+
+multiline_string = { "'''" ~ (!"'''" ~ ANY)* ~ "'''" }
+root = _{ multiline_string | inline_comment }
+code_block = { multiline_string | code_line+ }
```
Not the right approach -- a code block is:
- Not a doc block, followed by non-newline characters then a newline OR
- Not a doc block, followed by non-newline characters  then EOI OR
- non-newline characters followed by a newline OR
- a multi-line string

bjones1 commented:
> I put my change into the python pest file, for now it can well see inline comment in multiligne string as code block in this case ''' # this in not a comment ''', but not is this case : print(''' # this is not a comment '''). I aslo tried to revert my previous wrong commits.

Progress -- see my review comments.

bjones1 commented:
I see progress, but nothing from what we discussed in my office. I want a fully-working PR ready before we meet tomorrow.

williamaguessy commented:
Here is the change we discussed during our meeting but it's not working for the moment. It only recognise multiligne string but not normal code block anymore

williamaguessy commented:
in fact it detect only the multiline string evne if it is one of the two other case. The definition isn't complete

williamaguessy commented:
I have made the changes to handle single and double quotes with backslashes, and I have also added support for parsing both ''' and """ multiline strings. Initially, there was a conflict between the two rules because the parser interpreted the first two double quotes as the start of a multiline backslash. To resolve this, I replaced a * with a +. multiline_string_backslash = { 
"\"" ~ (("\\" ~ ANY) | (!( "\"" | NEWLINE ) ~ ANY))**+** ~ "\"" 
}
